internal/stream_fetch_helper.go

```diff
@@ -57,3 +58,28 @@ func downloadAndDecompressStream(backup Backup, writeCloser io.WriteCloser) erro
 	}
 	return newArchiveNonExistenceError(fmt.Sprintf("Archive '%s' does not exist.\n", backup.Name))
 }
+
+func downloadAndDecompressStreamParts(backup Backup, writeCloser io.WriteCloser, fileNames []string) error {
+	defer utility.LoggedClose(writeCloser, "")
+
+	decompressor := compression.FindDecompressor(filepath.Ext(fileNames[0])[1:])
+	if decompressor == nil {
+		return newUnknownCompressionMethodError()
+	}
+	for _, fileName := range fileNames {
```

Comment:
This process is synchronous, have you considered adding parallelism? For example, add an option to pre-fetch the next archive while doing the DecompressDecryptBytes of the current one?